"Post exporter" is stuck on amount of incidents > 10

**Describe the bug**

In new Camunda exporter we have the Tasks part that is doing some background work, including post-export processing of incidents and related entities (class `IncidentUpdateTask`). When many incidents are created in small amount of time and too many incidents come for processing in one batch, post export got stuck with exception and no new incidents can be processed.

**To Reproduce**

1. Run `StandaloneCamunda` with profiles `broker,operate,dev,dev-data` (e.g. [Operate `make env-up`](https://github.com/camunda/camunda/blob/64784feafdc11d2ab51b7d9f8905d09e94c6d13e/operate/Makefile#L4) can be used) OR generate many incidents fast (e.g. 100).
2. Wait for dev data being generated, observe the logs and Operate Dashboard.

**Expected behavior**

Amount of incidents continues growing more and more. Logs are not showing any `post export` related exceptions.

**Current behavior**

Incident processing is blocked, so only part of incidents are visible in Operate UI. Logs are showing reoccurring exceptions.

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
22:40:25.753 [] [exporter-camunda-p1-tasks-4] [] WARN  io.camunda.zeebe.broker.exporter.camunda - Error occurred while performing a background task; operation will be retried
java.util.concurrent.CompletionException: io.camunda.zeebe.exporter.api.ExporterException: Failed to fetch incidents associated with post-export updates; it's possible they are simply not visible yet. Missing incident IDs: [[2251799813689610, 2251799813688588, 2251799813688322, 2251799813691655, 2251799813688856, 2251799813689371, 2251799813689884, 2251799813687070, 2251799813692178, 2251799813694482, 2251799813693207, 2251799813687336, 2251799813692456, 2251799813690412, 2251799813689645, 2251799813690148, 2251799813690686, 2251799813689919, 2251799813687602, 2251799813690447, 2251799813687874, 2251799813692229, 2251799813690183, 2251799813691224, 2251799813688412, 2251799813688144, 2251799813690960, 2251799813693268, 2251799813691752, 2251799813689194, 2251799813694571, 2251799813691488, 2251799813690721, 2251799813688678, 2251799813691259, 2251799813688944, 2251799813690995, 2251799813685622, 2251799813687158, 2251799813694582, 2251799813691787, 2251799813685644, 2251799813687693, 2251799813689229, 2251799813689742, 2251799813687424, 2251799813685633, 2251799813691523, 2251799813689478, 2251799813685657, 2251799813687964, 2251799813693085, 2251799813692050, 2251799813688232, 2251799813690280, 2251799813686953, 2251799813689513, 2251799813685679, 2251799813690016, 2251799813685668, 2251799813685690, 2251799813690554, 2251799813689777, 2251799813688500, 2251799813694645, 2251799813690315, 2251799813690828, 2251799813688768, 2251799813690051, 2251799813686980, 2251799813685702, 2251799813694662, 2251799813687514, 2251799813693146, 2251799813689052, 2251799813691356, 2251799813690589, 2251799813685726, 2251799813687248, 2251799813693392, 2251799813685713, 2251799813691092, 2251799813687784, 2251799813690863, 2251799813691620, 2251799813689336, 2251799813689087, 2251799813691391, 2251799813688054, 2251799813691127]]
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:674) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:662) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.thenApplyAsync(CompletableFuture.java:2210) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.thenApplyAsync(CompletableFuture.java:144) ~[?:?]
	at io.camunda.exporter.tasks.ReschedulingTask.run(ReschedulingTask.java:55) ~[classes/:?]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:317) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[?:?]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: io.camunda.zeebe.exporter.api.ExporterException: Failed to fetch incidents associated with post-export updates; it's possible they are simply not visible yet. Missing incident IDs: [[2251799813689610, 2251799813688588, 2251799813688322, 2251799813691655, 2251799813688856, 2251799813689371, 2251799813689884, 2251799813687070, 2251799813692178, 2251799813694482, 2251799813693207, 2251799813687336, 2251799813692456, 2251799813690412, 2251799813689645, 2251799813690148, 2251799813690686, 2251799813689919, 2251799813687602, 2251799813690447, 2251799813687874, 2251799813692229, 2251799813690183, 2251799813691224, 2251799813688412, 2251799813688144, 2251799813690960, 2251799813693268, 2251799813691752, 2251799813689194, 2251799813694571, 2251799813691488, 2251799813690721, 2251799813688678, 2251799813691259, 2251799813688944, 2251799813690995, 2251799813685622, 2251799813687158, 2251799813694582, 2251799813691787, 2251799813685644, 2251799813687693, 2251799813689229, 2251799813689742, 2251799813687424, 2251799813685633, 2251799813691523, 2251799813689478, 2251799813685657, 2251799813687964, 2251799813693085, 2251799813692050, 2251799813688232, 2251799813690280, 2251799813686953, 2251799813689513, 2251799813685679, 2251799813690016, 2251799813685668, 2251799813685690, 2251799813690554, 2251799813689777, 2251799813688500, 2251799813694645, 2251799813690315, 2251799813690828, 2251799813688768, 2251799813690051, 2251799813686980, 2251799813685702, 2251799813694662, 2251799813687514, 2251799813693146, 2251799813689052, 2251799813691356, 2251799813690589, 2251799813685726, 2251799813687248, 2251799813693392, 2251799813685713, 2251799813691092, 2251799813687784, 2251799813690863, 2251799813691620, 2251799813689336, 2251799813689087, 2251799813691391, 2251799813688054, 2251799813691127]]
	at io.camunda.exporter.tasks.incident.IncidentUpdateTask.getPendingIncidentsBatch(IncidentUpdateTask.java:472) ~[classes/:?]
	at io.camunda.exporter.tasks.incident.IncidentUpdateTask.processNextBatch(IncidentUpdateTask.java:81) ~[classes/:?]
	at io.camunda.exporter.tasks.incident.IncidentUpdateTask.execute(IncidentUpdateTask.java:73) ~[classes/:?]
	at io.camunda.exporter.tasks.ReschedulingTask.run(ReschedulingTask.java:44) ~[classes/:?]
	... 7 more
```

</p>
</details>

**Environment:**

- Zeebe Version: 8.7.0-alpha2
- Configuration: `dev-data` profile is activated

--- a/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
+++ b/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java
@@ -374,6 +374,7 @@ public final class ElasticsearchIncidentUpdateRepository implements IncidentUpda
         .allowNoIndices(true)
         .ignoreUnavailable(true)
         .sort(s -> s.field(f -> f.field(IncidentTemplate.KEY)))
+        .size(incidentIds.size())
         .build();
   }
 

--- a/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/OpenSearchIncidentUpdateRepository.java
+++ b/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/OpenSearchIncidentUpdateRepository.java
@@ -429,6 +429,7 @@ public final class OpenSearchIncidentUpdateRepository implements IncidentUpdateR
         .allowNoIndices(true)
         .ignoreUnavailable(true)
         .sort(s -> s.field(f -> f.field(IncidentTemplate.KEY)))
+        .size(incidentIds.size())
         .build();
   }
 

--- a/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/incident/IncidentUpdateRepositoryIT.java
+++ b/zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/tasks/incident/IncidentUpdateRepositoryIT.java
@@ -49,11 +49,14 @@ import io.camunda.zeebe.exporter.api.ExporterException;
 import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
 import io.camunda.zeebe.test.util.junit.AutoCloseResources;
 import io.camunda.zeebe.test.util.junit.AutoCloseResources.AutoCloseResource;
+import io.camunda.zeebe.test.util.junit.RegressionTest;
 import io.camunda.zeebe.test.util.testcontainers.TestSearchContainers;
 import java.io.IOException;
 import java.time.Duration;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
@@ -286,6 +289,30 @@ abstract sealed class IncidentUpdateRepositoryIT {
           .hasSize(1)
           .containsEntry(
               "1", new IncidentDocument("1", incidentTemplate.getFullQualifiedName(), expected));
+    }
+
+    @RegressionTest("https://github.com/camunda/camunda/issues/25968")
+    void shouldReturnMoreThanOnePage() throws PersistenceException {
+      // given
+      final var repository = createRepository();
+      final List<String> ids = new ArrayList<>();
+      final Map<String, IncidentDocument> expected = new HashMap<>();
+      for (int i = 0; i < 20; i++) {
+        final var id = String.valueOf(i);
+        final var entity = createIncident(i);
+        ids.add(id);
+        expected.put(id, new IncidentDocument(id, incidentTemplate.getFullQualifiedName(), entity));
+      }
+
+      // when
+      final var documents = repository.getIncidentDocuments(ids);
+
+      // then
+      assertThat(documents)
+          .succeedsWithin(REQUEST_TIMEOUT)
+          .asInstanceOf(InstanceOfAssertFactories.map(String.class, IncidentDocument.class))
+          .hasSize(20)
+          .containsExactlyEntriesOf(expected);
     }
 
     private IncidentEntity createIncident(final long key) throws PersistenceException {

